tests/test_model_builders.py

```diff
@@ -0,0 +1,780 @@
+# ==============================================================================
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import unittest
+
+import lightgbm as lgbm
+import numpy as np
+import xgboost as xgb
+from sklearn.datasets import (
+    load_breast_cancer,
+    load_iris,
+    make_classification,
+    make_regression,
+)
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+
+import daal4py as d4p
+from daal4py.sklearn._utils import daal_check_version
+
+try:
+    import catboost as cb
+
+    cb_available = True
```

Comment:
i think we might end up in cases then lightgbm and xgboost will not be avaialable so would have same  skipps. 

The question, how we will be forcing actual tests execution, so we will not endup in state there steps will be green, but nothing will be tested. 

We can merge this one and discuss how to handle this

Reply:
Yes, I was thinking about it. I first had a comment in `requirements-test.txt`. Looked like this

```text
catboost==1.2.2; python_version <= '3.11' ; FIXME: Add to 3.12 when supported
shap==0.42.1; python_version <= '3.11'     ; FIXME: Add to 3.12 when supported
```

But I did not like the idea of `FIXME` comments in requirements files. For now, I will just add a reminder for myself.

Reply:
This is not a problem- fix me is fine and I would expect that with dependencies updates those restrictions would be validated by person integrating.

But the question on validation - package install might fail or done incorrectly across all platforms and we will not catch this because ci would be green.

Reply:
First, in this case I'm adding the `FIXME` comments back.

Second, I see the point. We might have to include such a testing gap in the release notes.
Something that says "Functionality dependent on SHAP and CatBoost packages not fully tested"

Alternatively, we could issue warnings on 3.12 if CatBoost functionality is used (like model building from CB models).
(Same for SHAP, but that is really only used in testing, so the user would never get any warnings.)